components/ntask/org.wso2.carbon.ntask.core/src/main/java/org/wso2/carbon/ntask/core/impl/standalone/StandaloneTaskManager.java

```diff
@@ -54,6 +54,15 @@ public void scheduleTask(String taskName) throws TaskException {
         }
     }
 
+    /**
+     * This method is introduced to be used with the ClusteredTaskManager with Quartz clustering, and is not supported
+     * in standalone mode.
+     */
+    @Override
+    public void scheduleTask(String taskName, boolean requestRecovery) throws TaskException {
+        throw new TaskException("Requesting recovery is not supported in standalone mode", null);
```

Comment:
Isn't it a bad design to have a method required by only a single concrete class in the generic interface?